src/finchlite/finch_assembly/nodes.py

```diff
@@ -714,6 +719,26 @@ def __call__(self, prgm: AssemblyNode):
                     f"{body_code}\n"
                 )
                 return None
+            case Function(TaggedVariable(Variable(func_name, return_t), id), args, body):
+                ctx_2 = self.subblock()
+                arg_decls = []
+                for arg in args:
+                    match arg:
+                        case TaggedVariable(Variable(name, t), id):
+                            arg_decls.append(f"{name}_{id}: {qual_str(t)}")
+                        case _:
+                            raise NotImplementedError(
+                                f"Unrecognized argument type: {arg}"
+                            )
+                ctx_2(body)
+                body_code = ctx_2.emit()
+                feed = self.feed
+                self.exec(
+                    f"{feed}def {func_name}_{id}({', '.join(arg_decls)}) -> "
+                    f"{qual_str(return_t)}:\n"
+                    f"{body_code}\n"
+                )
+                return None
```

Comment:
Can we remove it and modify `case Function` to not to duplicate it all? So `match` statement there will know whether we have variable or tagged(variable).